cdap-api/src/main/java/co/cask/cdap/api/lineage/field/EndPoint.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.lineage.field;
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * An EndPoint in an operation represents the source or destination of the data,
+ * along with the namespace in which they exists. When namespace is not provided,
+ * the namespace for an EndPoint is considered to be same as the namespace in which
+ * program runs.
+ */
+public class EndPoint {
+  private final String namespace;
+  private final String name;
+
+  public EndPoint(@Nullable String namespace, String name) {
+    this.namespace = namespace;
+    this.name = name;
+  }
+
+  /**
+   * @return the namespace name if it is explicitly provided while creating this EndPoint,
+   * otherwise {@code null} is returned
+   */
+  @Nullable
+  public String getNamespace() {
+    return namespace;
+  }
+
+  /**
+   * @return the name of the {@link EndPoint}
+   */
+  public String getName() {
+    return name;
+  }
+
+  /**
+   * Return the EndPoint as defined by the provided name.
+   *
+   * @param name the name of the EndPoint
+   * @return the EndPoint
+   */
+  public static EndPoint of(String name) {
+    return of(null, name);
+  }
+
+  /**
+   * Return the EndPoint as defined by the provided namespace and name.
+   *
+   * @param namespace the name of the namespace. EndPoint is associated with the
+   *                  namespace when it is non {@code null}, otherwise
+   *                  the namespace in which program runs is considered as the
+   *                  namespace for the EndPoint
+   * @param name the name of the EndPoint
+   * @return the EndPoint
+   */
+  public static EndPoint of(@Nullable String namespace, String name) {
```

Comment:
if the namespace is provided, it should not be null. You have another constructor for that case.

Reply:
fixed.